accepted/00074-retracted-updates-support.md

```diff
@@ -0,0 +1,363 @@
+synced in the past with `status=final`). We need to make sure reposync spots
+this change and maybe we also need to update some caches (system available
+updates are cached (`rhnServerNeededCache`?)).
+
+Analyze the case, when a stable patchh gets retracted and then stable again in SCC.
+
+### Java backend
+
+We need to enhance the `Errata` class with the `status` attribute, for which we
+will implement an `enum`.
+
+### Repository generating
+Enhance the `UpdateInfoWriter` class, so that it includes the `status` attribute
+in the output XML. Currently the `status` output is hardcoded to `final`.
+
+## Inter Server Sync
+Make sure that the patch status gets propagated correctly in the ISS slaves.
+
+
+## Iteration 2 - Enhance UI and XMLRPC
+
+In this iteration, we add the support to user-facing parts of SUMA.
+
+### UI
+
+The following pages must be updated:
+
+- *Patches -> All -> detail*:
+  information about the status. Given the fact we only have 2 states now
+  (`retracted`/`final`), this can be a boolean information telling whether the
+  patch is retracted or not.
+
+- *Patches -> Manage Patches*: 
+  We should implement a checkbox, allowing the user to change the status
+  attribute (although the change should be disabled for vendor patches). We need
+  to take into account the possible consequences of modifying a patch (maybe
+  regenerating some data (`rhnServerNeededCache`, regenerating repodata of all
+  involved channels)).
+
+- *Patches -> All*: an icon in the list, if the patch is retracted.
+
+- *Patches -> Relevant*: a retracted patch should not be displayed
+  in this list at all.
+
+- *System detail -> Software -> Packages -> List*:
+  Installed packages contained in retracted patches should be highlighted (icon,
+  possibly accompanied by red text).
+
+- *System detail -> Software -> Packages -> Upgrade*:
+  Package upgrades that are part of the retracted patch should not be
+  displayed. The highest ("non-retracted") update of the package should be
+  displayed.
+
+- *System detail -> Software -> Packages -> Install*:
+  Don't display package versions that are part of a retracted patch.
+
+- New tab *System detail -> Software -> Packages -> Retracted*:
+  This screen should allow user to install or upgrade packages
+  contained in retracted patches and should contain huge warning about
+  possible consequences. The table should contain packages for both
+  installation and upgrade in the following manner:
+
+  - Installation: display the newest version of the "retracted"
+    package, if it's the newest version in the channel (i.e. there is
+    no non-retracted package that superseedes this one) and the
+    package is not installed on the system.
+
+  - Upgrade: for each package installed on the system, display its
+    retracted upgrade, if there is no superseeding non-retracted
+    upgrade in the channel.
+
+  - We should distinguish the install/upgrade rows graphically
```

Comment:
Having the column "Installed Package" as in the "Upgrade" tab, and leaving the cell empty for install rows should do it

Reply:
Good observation!